client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -274,8 +263,10 @@ public synchronized int getId(String subject, Schema schema)
   }
 
   @Override
-  public List<Integer> deleteSubject(Map<String, String> requestProperties, String subject)
+  public synchronized List<Integer> deleteSubject(
```

Comment:
this is the actual thread-safety issues, right?

Reply:
One of them, yes.